micrometer-core/src/main/java/io/micrometer/core/instrument/binder/logging/Log4j2Metrics.java

```diff
@@ -118,28 +137,32 @@ private MetricsFilter getOrCreateMetricsFilterAndStart(MeterRegistry registry) {
 
     @Override
     public void close() {
+        changeListeners.forEach(loggerContext::removePropertyChangeListener);
+
         if (!metricsFilters.isEmpty()) {
             Configuration configuration = loggerContext.getConfiguration();
-            LoggerConfig rootLoggerConfig = configuration.getRootLogger();
-            metricsFilters.values().forEach(rootLoggerConfig::removeFilter);
-
-            loggerContext.getConfiguration()
-                .getLoggers()
-                .values()
-                .stream()
-                .filter(loggerConfig -> !loggerConfig.isAdditive())
-                .forEach(loggerConfig -> {
-                    if (loggerConfig != rootLoggerConfig) {
-                        metricsFilters.values().forEach(loggerConfig::removeFilter);
-                    }
-                });
-
+            removeMetricsFilter(configuration);
             loggerContext.updateLoggers(configuration);
+
             metricsFilters.values().forEach(MetricsFilter::stop);
-            metricsFilters.clear();
```

Comment:
Should not we keep this and clear `metricsFilters` and also `changeListeners`?

Reply:
Ah, of course, that was a mistake 😓 I'll clear both of these out!